### PR TITLE
[language] fix some bugs in bytestring literals and use them in the stdlib

### PR DESCRIPTION
**Pull request by bob-wilson**
There were a few character strings (actually vector<u8>) in the stdlib that were described as having UTF-8 encodings. However, these are all limited to ASCII characters, so we can use the new bytestring literal syntax to specify the values. We want to avoid getting into Unicode complexities, at least for now.

That change exposed a problem with comment markers inside strings. The addition of byte string literals to Move means that we can have comment-starting characters that occur inside a string, e.g., "//". The code to identify and strip comments needs to understand the string syntax so that it can avoid treating those characters as comments when they are used inside strings.

Finally, while fixing that comment problem, I happened to notice an issue with escaped backslashes before the closing quote of a string. The previous code looked for a closing quote that was not immediately preceded by a backslash. But, that doesn't handle the case where there is an escaped backslash before the quote.

## Motivation

Fix bugs and make the stdlib code more readable

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Added tests for the bytestring bugs and ran them

### CODE REVIEW

File: language/stdlib/modules/lbr.move [hunk 1]
```diff
@@ -44,7 +44,7 @@ module LBR {
             true,    // is_synthetic
             1000000, // scaling_factor = 10^6
             1000,    // fractional_part = 10^3
-            x"4C4252" // UTF8-encoded "LBR" as a hex string
+            b"LBR"
```
Comment (tnowacki) [inline language/stdlib/modules/lbr.move:47]: Beautiful!

Comment (tnowacki) [reply language/stdlib/modules/lbr.move:47]: @tzakian @sblackshear 
Any other spots to clean up?

Comment (bob-wilson) [reply language/stdlib/modules/lbr.move:47]: I searched for all the hex string literals but I suppose I might have missed something.

Comment (tnowacki) [reply language/stdlib/modules/lbr.move:47]: No idea, just figured I'd ask since @tzakian had talked about having to go onto a website to get the encoding

**Pull request closed without merging**